### PR TITLE
feat(docs):  reorganize docs

### PR DESCRIPTION
**Pull request by vwheeler63**
0.  Executive Summary:

    - This PR is a MAJOR re-organization of much of the introductory content of the documentation -- the material new LVGL programmers need early in their learning sequence to get up and coding with LVGL, and having wins with it, as quickly as possible.

    - In my opinion, the quality of the documentation from the viewpoint of a programmer new to LVGL is now about 3X clearer and more usable than what it was.  Many errors and ambiguities were corrected with coherent, clear explanations in a logical order.

    - Technical details on many topics are now in one place and easy to find.  Between Gábor and I, we found what must have been 8-10 different major [and important] topics that were scattered in 2, 3 and sometimes 4 different places, that are now consolidated in one place, and in many cases re-written or at least thoroughly edited for clarity.

    - Here is where you can see what it will look like:  http://crystal-clear-research.com/for_gabor/demo/docs.lvgl.io/master/index.html    Note that EXAMPLES and VERSION numbers do not work on this web server, as it needs to be on the actual documents host for these to work.  Also "Edit on GitHub" links will not work yet for the docs that have moved until this PR is committed.
    
    - In PR 7115, several improvements were discussed and they are implemented herein.  Details of these changes listed in `#11` below.
    
    - Details are below.

1.  The docs directory structure was reorganized to give each topic exactly one place to go, not 2, or 3 or 4 as was previously the case (and happened).

    The old structure invited certain topics to go into multiple locations (e.g. full explanations of how to set up a project, and about 8 other topics that are now consolidated).  This structure clarifies for the writer/editor where new topics (or updates to old ones) should go.

    There are now 2 major sections of the documentation from the viewpoint of content (not the table of contents, though they are related).  A)  introductory material that orients new programmers (./docs/intro/), and B) reference material (./docs/details/), in which each topic is fully covered and in one place.  (No material is in the introductory material that is not also in the reference material.  And introductory material mostly refers to the reference material for details.)

    The directory structure reflects these sections, and is also reflected in the HTML URL paths that are generated.

    I am hopeful that this directory structure and new location of topics will be self-explanatory to programmers adding to the documentation in the future.  But I will be adding some guidelines in the ./docs/README.md just in case it isn't perfectly self-explanatory.

2.  In the Introduction, Overview, Getting Started, Adding LVGL to Your Project, Modules and Base Widget sections, anywhere from 1/3 to 2/3 of the material was majorly re-written (consolidated and unified) into a coherent logical whole in clear English.  Basics such as the relationship between Screens, Displays and Widgets is far, FAR better explained now and will give a much more helpful "springboard" to get programmers new to LVGL up and winning with LVGL as fast as possible.

    Each section was designed to answer the questions:  "What is it?", "What does it do?", and "How does it do it?" as early as possible to make it easier for anyone (beginner and experienced) users to find what they are looking for and quickly get up to speed with it.  (Though there was good material before, there were places where this quality was lacking.)

    The result of `#1` and `#2` above is all-around WAY better documentation -- it is easier to get oriented for programmers new to LVGL, and easier to find things for experienced LVGL programmers.

4.  The term "Widget" was being badly conflated with the term "object" and "obj".  This undoubtedly propagated from the fact that the Base Widget's data type was 'lv_obj_t'.  The damage done caused the term "object" to be ubiquitous in the documentation when it meant "Widget".

    There are legitimate times when the term "object" should be used:  Display object, cursor object, and many other types of objects that are NOT widgets.  But the term "object", and "obj", "obj1", "obj2" etc. where it meant "Widget" was converted to "Widget", "widget1", "widget2", etc..  (There were probably close to a thousand such instances of this.  Note that I was VERY careful not to miss any instances where "object" was the correct term.)

    This included significant edits to comments in 3 code files:
    - ./scripts/style_api_gen.py  which generates ./docs/details/base-widget/styles/style-properties.rst
    - ./src/core/lv_group.h  (terms match updated documentation)
    - ./src/misc/lv_event.h  (terms match updated documentation)

5.  Consolidated (unified) content previously scattered in 2-4 places:

    - setting up LVGL in your project,
    - LVGL basics coverage of objects, widgets, events, parts, states,
        styles, themes,
    - base object,
    - calls used to create and modify widgets,
    - layers,
    - display,
    - input devices,
    - widget groups,
    - coordinates,
    - widget size,
    - widget position,
    - widget alignment,
    - several others I did not record.

6.  Thanks to the versatility of reStructuredText and Sphinx (our primary documentation-generation logic), the Table of Contents (TOC, that makes up page links and the navigation-tree) has ZERO to do with the underlying directory structure.  As far as it is concerned, the path to the documents is irrelevant.  It gives full control to the writer/editor as to how the TOC is organized.

7.  For reviewers, unfortunately, this is one of those cases where studying the CHANGES isn't going to do much good where there was a lot of re-writing, because of the extent of those changes is going to hide what really happened.  For the consolidated (unified) content, some of it was re-written, and some of it that was actually duplicate was disposed of.

    Example:  [ Introduction > Overview > LVGL Basics ]  was a major reorganization of this description of "basics", which was previously scattered.  It is now together in one, coherent whole, in the sequence needed for the reader to get an accurate understanding of the basics.  The various sub-sections of this page (down to "Events") now pull together this important, orienting material in an accurate, clear and in-sequence way for readers new to LVGL.

    Instead, instead of trying to decipher "what happened" using the DIFFs (where there was a lot of re-writing), use this link to see the end result of the reorganization.

    http://crystal-clear-research.com/for_gabor/demo/docs.lvgl.io/master/index.html

    That said, you can rest assured that I was very careful to preserve everything except things which actually WERE duplicate or redundant.  Consolodating topics revealed these things so the actual SIZE of the documentation is now exactly what it was minus only the small amount that was removed for redundancy.

8.  There are 2 legitimate Sphinx warnings that are generated:

    - <tmpdir>/intro/index.rst: WARNING: document isn't included in any toctree
    - <tmpdir>/details/index.rst: WARNING: document isn't included in any toctree

    This is intentional.  The reason these are legitimate is that this gives a way to:

    - have the Introduction and Reference sub-pages immediately visible in the top-level table of contents without hiding them in a sub-page.  So this:

            Introduction

              - Introduction   <-- link

            Details

              - Details        <-- link

            ...

      turns into this:

            Introduction

              - Introduction
              - Overview
              - Getting Started
              - Add LVGL to Your Project

            Details

              - Configuration
              - Modules
              - Base Object
              - Widgets
              - Other Components
              - Examples
              - Debugging
              - Integration and Drivers
              - 3rd-Party Libraries

            ...

      Note that these are direct links to sub-pages.  This is an intentional departure from our usual practice of capturing a group of sub-pages under an INDEX.RST file which generates an `index.html` file, but winds up hiding the sub-pages in the higher-level TOC.

      The `index.rst` file would not even be needed except when a user types THIS into his browser:

          https://docs.lvgl.io/master/intro/

      the web server will look for an `index.html` file and will not find it, and serve a 404 (file not found) error instead. So I added the `index.rst` file (that does not appear in this TOC) in order to generate the needed `index.html`.  (I realize there are ways to get web servers to serve up other "default" pages in a directory, but this avoids the need for that.)  This way, when the user tries the above he will get a nice-looking page that contains the sub-pages, and look like this:

            Introduction

              - Introduction
              - Overview
              - Getting Started
              - Add LVGL to Your Project

      The presence of this index.rst file NOT connected to any TOC generates the warning.

9.  Besides re-writing many parts of the beginning half of the documentation (for clarity, and use of standard English), some parts were found (and corrected) that were just plain wrong.  For example:

        https://docs.lvgl.io/master/overview/display.html#background

    stated that every `lv_display_t` object has a

        - background color
        - background image, and
        - opacity properties.

    This is not even vaguely correct, and probably occurred because the writer was confusing Screens and Displays.  This was thoroughly corrected, and the material re-located where it belonged:  with the (now thorough) description of Screens under Base Object (not under Display, though the use of Screens as layers is correctly covered under Display).  There were a handful of others (e.g. in cases where there are multiple displays, the Default Display was reported to be the LAST one created, but it is actually the FIRST one created according to the C code).

10.  `@example` Doxygen command was removed from ./src/widgets/label/lv_label.h  because it does not do what the writer expected, and for this reason it is not one of the supported Doxygen commands.

11. Improvements from PR #7115:

    - ToC reorganized per guidance from Gábor.
    - Confirmation:  recent changes to docs (in last several weeks) are all present and preserved.
    - Overview renamed to "LVGL Basics" as is the 2nd button on landing page (you may have to do a "hard refresh" to see the button update).
    - LVGL Basics ToC grouped to address "flat ToC" and to help reader understand context.
    - Redundancy with "Getting Started" and "Overview" solved by renaming Overview.
    - Modules => Main Components
    - Sphinx-redirects implemented for all top-level ToC pages that moved.  See links above.

      To see examples, try these links (that redirect):

      - http://crystal-clear-research.com/for_gabor/demo/docs.lvgl.io/master/get-started/index.html
      - http://crystal-clear-research.com/for_gabor/demo/docs.lvgl.io/master/integration/index.html
      - http://crystal-clear-research.com/for_gabor/demo/docs.lvgl.io/master/porting/index.html
      - http://crystal-clear-research.com/for_gabor/demo/docs.lvgl.io/master/overview/index.html
      - http://crystal-clear-research.com/for_gabor/demo/docs.lvgl.io/master/layouts/index.html
      - http://crystal-clear-research.com/for_gabor/demo/docs.lvgl.io/master/libs/index.html
      - http://crystal-clear-research.com/for_gabor/demo/docs.lvgl.io/master/others/index.html

    - Recent doc changes verified as included:
        - introduction of debugging/ page and sub-pages and subsequent enhancements + gdb_plugin.rst.
        - updates to .../integration/chip/nxp.rst
        - e470349 updates to links (made resilient to doc moves by converting to Sphinx hyperlink refs)
        -  added yocto to .../integration/os/yocto/ pages
        - 30193c1 enhancements to .../integration/chip/stm32.rst + new nema_gfx.rst
        - 0efa5f3 enhancements to .../integration/driver/opengles.rst
        - 5081fbc enhancements to .../integration/os/nuttx.rst
        - ce52267 removal of erroneous `<strong>...</strong>` HTML tags in .../widgets/scale.rst
        - new .../integration/driver/display/st_ltdc.rst
        - new .../libs/svg.rst
        - new .../others/font_manager.rst
        - 032c2c3 sorting of sub-pages in several `index.rst` files.
        - moving of log.rst and profiler.rst and others vg_lite_tvg.rst under .../debugging/
        - 07854b1 adding TSC Images section to nema_gfx.rst

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.  DONE!
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.  n/a
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.  n/a
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).  n/a
- Run `scripts/code-format.py` (`astyle v3.4.12` needs to installed by running `cd scripts; ./install_astyle.sh`) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).  n/a
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.  Done.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.  Done.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Comment by kisvegabor** [comment]
I like it! Let"s try to merge it as soon as possible and add smaller changes in new PRs. 

cc @liamHowatt @XuNeo @FASTSHIFT

**Comment by vwheeler63** [comment]
> I like it! Let"s try to merge it as soon as possible and add smaller changes in new PRs.

Excellent!

**Comment by vwheeler63** [comment]
Force pushed to update after re-base (to resolve conflicts).

File: docs/details/other-components/obj_id.rst [hunk 1]
```diff
@@ -33,26 +33,26 @@ Below APIs needed to be implemented and linked to lvgl.
 
 .. code-block:: c
 
-    void lv_obj_set_id(lv_obj_t * obj, void * id);
-    void lv_obj_assign_id(const lv_obj_class_t * class_p, lv_obj_t * obj);
-    void lv_obj_free_id(lv_obj_t * obj);
-    const char * lv_obj_stringify_id(lv_obj_t * obj, char * buf, uint32_t len);
+    void lv_obj_set_id(lv_obj_t * widget, void * id);
```
Comment (XuNeo) [inline docs/details/other-components/obj_id.rst:36]: Is this change intentional? It's inconsistent with source code.

**Review by XuNeo — APPROVED** [review]

File: docs/details/other-components/obj_id.rst (hunk 1 above)
Comment (vwheeler63) [reply docs/details/other-components/obj_id.rst:36]: > Is this change intentional? It's inconsistent with source code.

Yes, it was intentional and coordinated with Gábor.  Reason:  we are both soon expecting the type name `lv_obj_t` to change to `lv_widget_t` or something close to that.  Gábor gave me the "green light" to start using the term "widget" in this context, including in function arguments.

Comment (XuNeo) [reply docs/details/other-components/obj_id.rst:36]: Hi @kisvegabor , 
Could you confirm? When we upgrade to lvgl v9, we agreed to not mess with `lv_obj_` prefix, in order to avoid massive API break.
cc @FASTSHIFT @W-Mai @terry0012

Comment (kisvegabor) [reply docs/details/other-components/obj_id.rst:36]: Confirmed. 

> When we upgrade to lvgl v9, we agreed to not mess with lv_obj_ prefix, in order to avoid massive API break.

IIRC, we agreed to not do it in v9. :smile: IMO we should slowly transition towards the `widget`name and we can do it without API breaks by adding backward compatibility defines in `lv_api_map.h`.

I don't have an exact plan when and how to introduce this refactoring, but once the docs were reworked, I find it a good time to start using the "widget" name.

**Review by kisvegabor — APPROVED** [review]

**Comment by kisvegabor** [comment]
Huh, merged! Curiously waiting how it will work :crossed_fingers:

**Comment by yutotakano** [comment]
Literally just went through adding LVGL and getting things working yesterday, oh how much I wish these new docs were here then! 😅

Thank you for the PR! Came across it when many of my tabs started 404-ing after a reload.

**Comment by vwheeler63** [comment]
> There is a larger pack of warnings [here](https://github.com/lvgl/lvgl/actions/runs/11486315251/job/31968335879#step:11:41). Is it known?

I think so.  I will compare this list with the one I am getting here to make sure.  Some of them are expected and are [explained above](https://github.com/lvgl/lvgl/pull/7136#issue-2605375005).  A few appear to be additional warnings from a different version of `sphinx-build` as side effects of the ones explained above.  But at a glance, I need to take a closer look to make sure there aren't any bad links.

File: docs/details/other-components/obj_id.rst (hunk 1 above)
Comment (W-Mai) [reply docs/details/other-components/obj_id.rst:36]: I think lv_view_t is more shorter 🤔

Comment (FASTSHIFT) [reply docs/details/other-components/obj_id.rst:36]: I suggest launching a public vote: choose to keep `obj` or use `widget`, because LVGL has such a large audience, and this core API change will affect everyone.

**Comment by vwheeler63** [comment]
> There is a larger pack of warnings [here](https://github.com/lvgl/lvgl/actions/runs/11486315251/job/31968335879#step:11:41). Is it known?

... and a quite a few are a known set from `breathe` complaining.

**Comment by kisvegabor** [comment]
In general the docs seems working :tada: 

However, I found that the links are not redirected. Basically none of te links in the [README](https://github.com/lvgl/lvgl) are working.

**Comment by FASTSHIFT** [comment]
@vwheeler63 
I found a documentation error [here](https://docs.lvgl.io/master/details/integration/renderers/nxp_vglite.html). You may have confused my `vg_lite.rst` with NXP's vglite, my `vg_lite `rendering backend is completely separate from NXP's implemented vglite rendering backend, please recover my documentation.

cc @kisvegabor

**Comment by nicusorcitu** [comment]
> Can I rename lv_vglite to lv_nxp_vglite? Indicates that vglite GPU are specifically adapted to NXP versions to avoid confusion. @nicusorcitu
>
Historically, `lv_vglite` was added first, together with `lv_pxp` - all inside the the `draw/nxp` folder. At the moment of introducing it, was no place for confusions :)

Now, with the Xiaomi version of `lv_vg_lite` available, yes, we can think renaming one of them to be much cleaner.

We should create a separate issue to be addressed in lvgl 9.3.
@cosmindanielradu19 @cristian-stoica

**Comment by kisvegabor** [comment]
> Now, with the Xiaomi version of lv_vg_lite available, yes, we can think renaming one of them to be much cleaner.
We should create a separate issue to be addressed in lvgl 9.3.

I think we should orient towards picking the best parts of each implementation and keeping only one.

**Comment by vwheeler63** [comment]
> In general the docs seems working 🎉
> 
> However, I found that the links are not redirected. Basically none of te links in the [README](https://github.com/lvgl/lvgl) are working.

I'm on it...

**Comment by nicusorcitu** [comment]
> I think we should orient towards picking the best parts of each implementation and keeping only one superior version.
>
I believe that too. For LVGL10 I think we should start a discussion and unify the two implementations somehow.

**Comment by vwheeler63** [comment]
> I found a documentation error [here](https://docs.lvgl.io/master/details/integration/renderers/nxp_vglite.html). You may have confused my `vg_lite` with NXP's vglite.
> My `vg_lite` rendering backend is separate from NXP's implemented vglite rendering backend, please **RECOVER** my documentation.

cc:  @FASTSHIFT @kisvegabor 

Done in PR #7148.  I changed the title based on not understanding what it was.  I should have read the document.  Instead I did a web search and *thought* I was putting in the "official title" -- that it was actually titling a GPU as opposed to rendering software designed to utilize a GPU API.  I have not yet done any projects using a GPU, so this was based on a superficial understanding of that context.  My apologies.

**Comment by FASTSHIFT** [comment]
> > I found a documentation error [here](https://docs.lvgl.io/master/details/integration/renderers/nxp_vglite.html). You may have confused my `vg_lite` with NXP's vglite.
> > My `vg_lite` rendering backend is separate from NXP's implemented vglite rendering backend, please **RECOVER** my documentation.
> 
> cc: @FASTSHIFT @kisvegabor
> 
> Done in PR #7148. I changed the title based on not understanding what it was. I should have read the document. Instead I did a web search and _thought_ I was putting in the "official title" -- that it was actually titling a GPU as opposed to rendering software designed to utilize a GPU API. I have not yet done any firmware projects using a GPU, so this was based on a superficial understanding of that context. My apologies.

It's okay. Thank you so much for fixing it. There is also a little fix, I made a message in the PR https://github.com/lvgl/lvgl/pull/7148.
